include/zephyr/net/buf.h

```diff
@@ -2414,6 +2414,23 @@ size_t net_buf_append_bytes(struct net_buf *buf, size_t len,
 			    const void *value, k_timeout_t timeout,
 			    net_buf_allocator_cb allocate_cb, void *user_data);
 
+/**
+ * @brief Compare data with a net_buf's content
+ *
+ * @details Compare data with a content of a net buf. If needed, traverse
+ * through multiple buffer fragments.
+ *
+ * @param buf Network buffer
+ * @param offset Starting offset to compare from
+ * @param data Data buffer for comparison
+ * @param data_len Length of the data buffer
+ * @param len Number of bytes to compare
+ *
+ * @return true if the data matches, false otherwise
+ */
+bool net_buf_data_cmp(struct net_buf *buf, size_t offset, void *data,
+		      size_t data_len, size_t len);
```

Comment:
Traditional "compare" functions return <0, 0 or >0, maybe rename it to `net_buf_data_eq` or convert the return type to `int` based on the first different byte found (similar to `strcmp`).

Reply:
I agree, it might be useful to know how many bytes matched.

Reply:
@pdgendt @jukkar The function now returns a number of bytes matching.